packages/react-components/react-tree/src/components/TreeItem/useTreeItem.ts

```diff
@@ -15,23 +16,12 @@ import { treeDataTypes } from '../../utils/tokens';
  * @param props - props from this instance of TreeItem
  * @param ref - reference to root HTMLElement of TreeItem
  */
-export function useTreeItem_unstable<Value = string>(
-  props: TreeItemProps<Value>,
-  ref: React.Ref<HTMLDivElement>,
-): TreeItemState {
+export function useTreeItem_unstable(props: TreeItemProps, ref: React.Ref<HTMLDivElement>): TreeItemState {
   const contextLevel = useTreeContext_unstable(ctx => ctx.level);
 
-  const id = useId('fui-TreeItem-', props.id);
+  const value = useId('fuiTreeItemValue-', props.value?.toString());
```

Comment:
If a component gets unmounted (for example, in case of virtualization) and `props.value` is missing ➡️ new `id` i.e. value will be regenerated. Did you consider this case? Is it okay?

Reply:
(if it's not important, can't we go with just id's instead of data attributes?)

Reply:
> (if it's not important, can't we go with just id's instead of data attributes?)

This seems to be a problem of coherence from our users perspective, we used to use `id` for that, and multiple partners have reported that the `id` in the flat tree case is confusing, as they don't expect the `id` property provided to the hook method to be a global `id` of an element, but a local `id`

Reply:
> Did you consider this case? Is it okay?

Yes, I've considered and it is kind of ok, the only thing that'll happen is if the user does not provide a proper value than as soon as the user closes the tree nested parts of it will also be closed (and we'll accumulate useless `ids` on the opened items (don't see this as a big issue though)